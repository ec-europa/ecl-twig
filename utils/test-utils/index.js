const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
