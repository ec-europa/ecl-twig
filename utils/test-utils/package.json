{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
