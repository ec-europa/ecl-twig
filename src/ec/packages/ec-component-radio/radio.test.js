import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import {
  dataDefault,
  dataInvalid,
  dataBinary,
  dataBinaryInvalid,
} from './demo/data';

describe('EC - Radio', () => {
  const template = '@ecl-twig/ec-component-radio/radio-group.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataDefault, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataDefault, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Invalid', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataInvalid)).resolves.toMatchSnapshot();
    });
  });

  describe('Binary', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataBinary)).resolves.toMatchSnapshot();
    });
  });

  describe('Binary invalid', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataBinaryInvalid)).resolves.toMatchSnapshot();
    });
  });
});
