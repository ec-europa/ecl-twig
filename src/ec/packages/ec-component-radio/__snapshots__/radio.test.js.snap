// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Radio Binary invalid renders correctly 1`] = `
<fieldset
  class="ecl-form-group ecl-form-group--radio ecl-radio__group--binary"
>
  <div
    class="ecl-help-block"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-feedback-message"
  >
    Error message for the group
  </div>
  <div
    class="ecl-radio"
  >
    <input
      checked=""
      class="ecl-radio__input"
      id="radio-binary-invalid-1"
      name=""
      type="radio"
      value="yes"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-invalid-1"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      Yes
    </label>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      class="ecl-radio__input"
      id="radio-binary-invalid-2"
      name=""
      type="radio"
      value="no"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-invalid-2"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      No
    </label>
  </div>
</fieldset>
`;

exports[`EC - Radio Binary renders correctly 1`] = `
<fieldset
  class="ecl-form-group ecl-form-group--radio ecl-radio__group--binary"
>
  <div
    class="ecl-help-block"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-radio"
  >
    <input
      checked=""
      class="ecl-radio__input"
      id="radio-binary-1"
      name=""
      type="radio"
      value="yes"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-1"
    >
      <span
        class="ecl-radio__box"
      />
      Yes
    </label>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      class="ecl-radio__input"
      id="radio-binary-2"
      name=""
      type="radio"
      value="no"
    />
    <label
      class="ecl-radio__label"
      for="radio-binary-2"
    >
      <span
        class="ecl-radio__box"
      />
      No
    </label>
  </div>
</fieldset>
`;

exports[`EC - Radio Default renders correctly 1`] = `
<fieldset
  class="ecl-form-group ecl-form-group--radio"
>
  <div
    class="ecl-help-block"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name=""
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box"
      />
      Luxembourg
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name=""
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box"
      />
      Belgium
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name=""
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-help-block ecl-radio__help ecl-help-block--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;

exports[`EC - Radio Default renders correctly with extra attributes 1`] = `
<fieldset
  class="ecl-form-group ecl-form-group--radio"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-help-block"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name=""
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box"
      />
      Luxembourg
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name=""
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box"
      />
      Belgium
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name=""
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-help-block ecl-radio__help ecl-help-block--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;

exports[`EC - Radio Default renders correctly with extra class names 1`] = `
<fieldset
  class="ecl-form-group ecl-form-group--radio custom-class custom-class--test"
>
  <div
    class="ecl-help-block"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-default-1"
      name=""
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-1"
    >
      <span
        class="ecl-radio__box"
      />
      Luxembourg
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-default-2"
      name=""
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-2"
    >
      <span
        class="ecl-radio__box"
      />
      Belgium
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-default-3"
      name=""
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label"
      for="radio-default-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-help-block ecl-radio__help ecl-help-block--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;

exports[`EC - Radio Invalid renders correctly 1`] = `
<fieldset
  class="ecl-form-group ecl-form-group--radio"
>
  <div
    class="ecl-help-block"
  >
    This is the group's helper text.
  </div>
  <div
    class="ecl-feedback-message"
  >
    This is the error message
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-1"
      checked=""
      class="ecl-radio__input"
      id="radio-invalid-1"
      name=""
      type="radio"
      value="lu"
    />
    <label
      class="ecl-radio__label"
      for="radio-invalid-1"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      Luxembourg
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-1"
    >
      Help text for option 1
    </div>
  </div>
  <div
    class="ecl-radio"
  >
    <input
      aria-describedby="helper-2"
      class="ecl-radio__input"
      id="radio-invalid-2"
      name=""
      type="radio"
      value="be"
    />
    <label
      class="ecl-radio__label"
      for="radio-invalid-2"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid"
      />
      Belgium
    </label>
    <div
      class="ecl-help-block ecl-radio__help"
      id="helper-2"
    >
      Help text for option 2
    </div>
  </div>
  <div
    class="ecl-radio ecl-radio--disabled"
  >
    <input
      aria-describedby="helper-3"
      class="ecl-radio__input"
      disabled=""
      id="radio-invalid-3"
      name=""
      type="radio"
      value="fr"
    />
    <label
      class="ecl-radio__label"
      for="radio-invalid-3"
    >
      <span
        class="ecl-radio__box ecl-radio__box--invalid ecl-radio__box--disabled"
      />
      France (disabled)
    </label>
    <div
      class="ecl-help-block ecl-radio__help ecl-help-block--disabled"
      id="helper-3"
    >
      Help text for option 3
    </div>
  </div>
</fieldset>
`;
